apps/gitness/src/components-v2/file-editor.tsx

```diff
@@ -34,18 +35,21 @@ export const FileEditor = ({ repoDetails, defaultBranch }: FileEditorProps) => {
   const [fileName, setFileName] = useState('')
   const [language, setLanguage] = useState('')
   const [originalFileContent, setOriginalFileContent] = useState('')
-  const [content, setContent] = useState(originalFileContent)
+  const [contentRevision, setContentRevision] = useState({ code: originalFileContent })
   const [view, setView] = useState<EditViewTypeValue>('edit')
   const [dirty, setDirty] = useState(false)
   const [isCommitDialogOpen, setIsCommitDialogOpen] = useState(false)
   const { selectedBranchTag } = useRepoBranchesStore()
+  const { theme } = useThemeStore()
+  // TODO: temporary solution for matching themes
+  const monacoTheme = (theme ?? '').startsWith('dark') ? 'dark' : 'light'
 
   const themeConfig = useMemo(
     () => ({
-      defaultTheme: 'dark',
+      defaultTheme: monacoTheme,
```

Comment:
Looks like you're having to repeat yourself in a bunch of places. It might make sense to make a theme-aware editor component that you can reuse. Or extract a custom hook to get the theme.

```typescript
import { useMemo } from 'react'

import { ThemeDefinition } from '@harnessio/yaml-editor'

import type { useThemeStore } from '../framework/context/ThemeContext.tsx'
import { themes } from '../pages/pipeline-edit/theme/monaco-theme.ts'

export interface UseThemeConfigPayload {
  defaultTheme: string
  themes: ThemeDefinition[]
}

export function useThemeConfig(): UseThemeConfigPayload {
  const { theme } = useThemeStore()

  return useMemo(
    () => ({
      defaultTheme: (theme || '').startsWith('dark') ? 'dark' : 'light',
      themes
    }),
    [theme]
  )
}

```

Then you can just call...
```typescript
const themeConfig = useThemeConfig()
```
in all the places you need it

Reply:
nice, will be done in the one of the follow up PRs.